github/github.go

```diff
@@ -336,7 +336,11 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 		return nil, err
 	}
 
-	defer resp.Body.Close()
+	defer func() {
+		// Drain and close the body to let the Transport reuse the connection
+		io.Copy(ioutil.Discard, resp.Body)
```

Comment:
> On successful queries (on errors ReadAll runs in CheckResponse), the
> Response.Body is not read all the way to the EOF as json.Decoder is used,
> which stops at the end of the object, which is probably followed by a \n.

> Since we know GitHub API responses only carry a single JSON object, draining
> before closing is free

I'm just curious, but have you tried doing something like `n, _ := io.Copy(ioutil.Discard, resp.Body)` here and dump the value of `n`? I'm curious if it's indeed 1 and contains a `\n`, or something else.